lib/chef/resource/chef_client_config.rb

```diff
@@ -0,0 +1,265 @@
+        default: ChefConfig::Config.etc_chef_dir
+
+      property :user, String,
+        description: "The user that should own the client.rb file and the configuration directory if it needs to be created."
+
+      property :node_name, String,
+        description: "The name of the node. This determines which configuration should be applied and sets the `client_name`, which is the name used when authenticating to a #{ChefUtils::Dist::Server::PRODUCT}. If this value is not provided #{ChefUtils::Dist::Infra::PRODUCT} will use the node's FQDN as the node name. Leaving this setting blank and letting the client assign the FQDN of the node as the node_name during each #{ChefUtils::Dist::Infra::PRODUCT} run is recommended."
+
+      property :chef_server_url, String,
+        description: "The URL for the #{ChefUtils::Dist::Server::PRODUCT}.",
+        required: true
+
+      # @todo Allow passing this as a string and convert it to the symbol
+      property :ssl_verify_mode, [Symbol, String],
+        equal_to: %i{verify_none verify_peer},
+        coerce: proc { |x| string_to_symbol(x) },
+        description: <<~DESC
+        Set the verify mode for HTTPS requests.
+
+        * Use :verify_none for no validation of SSL certificates.
+        * Use :verify_peer for validation of all SSL certificates, including the #{ChefUtils::Dist::Server::PRODUCT} connections, S3 connections, and any HTTPS remote_file resource URLs used in #{ChefUtils::Dist::Infra::PRODUCT} runs. This is the recommended setting.
+        DESC
+
+      property :formatters, Array,
+        description: "",
+        default: []
+
+      property :event_loggers, Array,
+        description: "",
+        default: []
+
+      property :log_level, Symbol,
+        description: "The level of logging performed by the #{ChefUtils::Dist::Infra::PRODUCT}.",
+        equal_to: %i{auto trace debug info warn fatal}
+
+      property :log_location, [String, Symbol],
+        description: "The location to save logs to. This can either by a path to a log file on disk `:syslog` to log to Syslog, `:win_evt` to log to the Windows Event Log, or `'STDERR'`/`'STDOUT'` to log to the *nix text streams.",
+        callbacks: {
+          "accepts Symbol values of ':win_evt' for Windows Event Log or ':syslog' for Syslog" => lambda { |p|
+            p.is_a?(Symbol) ? %i{win_evt syslog}.include?(p) : true
+          },
+        }
+
+      property :http_proxy, String,
+        description: "The proxy server to use for HTTP connections."
+
+      property :https_proxy, String,
+        description: "The proxy server to use for HTTPS connections."
+
+      property :ftp_proxy, String,
+      description: "The proxy server to use for FTP connections."
+
+      property :no_proxy, [String, Array],
+        description: "A comma-separated list or an array of URLs that do not need a proxy.",
+        coerce: proc { |x| x.is_a?(Array) ? x.join(",") : x },
```

Comment:
```suggestion
        coerce: proc { |x| x.is_a?(Array) ? x.join(",") : String(x) },
```

nitpick: this could end up nil

Reply:
I believe we only coerce if set so the value couldn't be nil.